tests/7zip_compatiblity_test.rs

```diff
@@ -0,0 +1,115 @@
+#![cfg(feature = "test-7zip")]
+//! This test compresses some random data with deflate64 using p7zip `7z` command and check decompression
+
+use std::ffi::OsString;
+use std::fs::File;
+use deflate64::Deflate64Decoder;
+use std::io::{Cursor, Read, Write};
+use std::mem::size_of;
+use std::process::{Command, Stdio};
+use proptest::proptest;
+use tempfile::TempDir;
+
+const TEST_FILE_NAME: &'static str = "test.file";
+const TEST_ZIP_NAME: &'static str = "test.zip";
+
+#[repr(packed)]
```

Comment:
```suggestion
#[repr(packed)]
#[derive(Default)]
```

You can then use `ZipLocalFileHeader::default()` for initialization instead of using `unsafe` code